core/src/subgraph/registrar.rs

```diff
@@ -668,10 +679,79 @@ fn create_subgraph_version(
         },
     });
 
+    let start_blocks: Option<Vec<EthereumBlockPointer>> = manifest
+        .data_sources
+        .clone()
+        .into_iter()
+        .map(|data_source| {
+            let source_address = data_source.source.address;
+            let block_number = match data_source
+                .source
+                .start_block
+                .and_then(|block_str| block_str.parse::<u64>().ok())
+            {
+                Some(number) => number,
+                None => {
```

Comment:
No, though originally I did use this for that purpose.  I've since made it more flexible; it doesn't end the iterator it just returns none for the current map then I filter out any `None` values in the next step: `.filter(|&n| n.is_some())`.